docs/source/guides/data.rst

```diff
@@ -0,0 +1,383 @@
+
+.. code-block:: python
+
+    # setup your data loader
+    test = DataLoader(...)
+
+    # test (pass in the loader)
+    trainer.test(test_dataloaders=test)
+
+--------------
+
+
+.. _sequences:
+
+
+***************
+Sequential Data
+***************
+
+Lightning has built in support for dealing with sequential data.
+
+
+Packed sequences as inputs
+==========================
+When using PackedSequence, do 2 things:
+
+1. Return either a padded tensor in dataset or a list of variable length tensors in the DataLoader collate_fn (example shows the list implementation).
+2. Pack the sequence in forward or training and validation steps depending on use case.
+
+.. testcode::
+
+    # For use in DataLoader
+    def collate_fn(batch):
+        x = [item[0] for item in batch]
+        y = [item[1] for item in batch]
+        return x, y
+
+    # In module
+    def training_step(self, batch, batch_nb):
+        x = rnn.pack_sequence(batch[0], enforce_sorted=False)
+        y = rnn.pack_sequence(batch[1], enforce_sorted=False)
+
+----------
+
+Truncated Backpropagation Through Time (TBPTT)
+==============================================
+There are times when multiple backwards passes are needed for each batch.
+For example, it may save memory to use Truncated Backpropagation Through Time when training RNNs.
+
+Lightning can handle TBPTT automatically via this flag.
+
+.. testcode::
+
+    from pytorch_lightning import LightningModule
+
+    class MyModel(LightningModule):
+
+        def __init__(self):
+            super().__init__()
+            # Important: This property activates truncated backpropagation through time
+            # Setting this value to 2 splits the batch into sequences of size 2
+            self.truncated_bptt_steps = 2
+
+        # Truncated back-propagation through time
+        def training_step(self, batch, batch_idx, hiddens):
+            # the training step must be updated to accept a ``hiddens`` argument
+            # hiddens are the hiddens from the previous truncated backprop step
+            out, hiddens = self.lstm(data, hiddens)
+            return {
+                "loss": ...,
+                "hiddens": hiddens
+            }
+
+.. note:: If you need to modify how the batch is split,
+    override :func:`~pytorch_lightning.core.LightningModule.tbptt_split_batch`.
+
+----------
+
+Iterable Datasets
+=================
+Lightning supports using IterableDataSets as well as map-style DataSets. IterableDataSets provide a more natural
+option when using sequential data.
+
+.. note:: When using an IterableDataSet you must set the ``val_check_interval`` to 1.0 (the default) or an int
+    (specifying the number of training batches to run before validation) when initializing the Trainer. This is
+    because the IterableDataSet does not have a ``__len__`` and Lightning requires this to calculate the validation
```

Comment:
```suggestion
.. note:: When using an IterableDataset you must set the ``val_check_interval`` to 1.0 (the default) or an int
    (specifying the number of training batches to run before validation) when initializing the Trainer. This is
    because the IterableDataset does not have a ``__len__`` and Lightning requires this to calculate the validation
```